pygmt/src/tilemap.py

```diff
@@ -0,0 +1,143 @@
+"""
+tilemap - Plot XYZ tile maps.
+"""
+from pygmt.clib import Session
+from pygmt.datasets.tile_map import load_tile_map
+from pygmt.helpers import (
+    GMTTempFile,
+    build_arg_string,
+    fmt_docstring,
+    kwargs_to_strings,
+    use_alias,
+)
+
+try:
+    import rioxarray
+except ImportError:
+    rioxarray = None
+
+
+@fmt_docstring
+@use_alias(
+    A="img_out",
+    B="frame",
+    C="cmap",
+    D="img_in",
+    E="dpi",
+    G="bit_color",
+    I="shading",
+    J="projection",
+    M="monochrome",
+    N="no_clip",
+    Q="nan_transparent",
+    # R="region",
```

Comment:
Was playing with the `zoom` parameter just now, and it seems that setting lower zoom levels (e.g. 0) with actually cause the area covered by `contextily.bounds2img` to be bigger than the original `region`. This might come as a surprise for users, so maybe we should set `no_clip` to True by default? This might require a bit of extra work to get correct.

Alternatively, we could also do the clipping in `pygmt.datasets.load_tile_map`.

Reply:
Added commit 17d07341eea751bc392cad3225c8a72b8ee87559 to ensure that:
- The plot is clipped to the bounding box region when `no_clip=False` (default)
- The plot extends to areas outside of the bounding box region when `no_clip=True`